src/main/java/org/traccar/handler/FilterHandler.java

```diff
@@ -142,22 +147,18 @@ private boolean skipAttributes(Position position) {
 
     private boolean filter(Position position) {
 
-        StringBuilder filterType = new StringBuilder();
-
-        Position last = null;
-        if (Context.getIdentityManager() != null) {
-            last = Context.getIdentityManager().getLastPosition(position.getDeviceId());
+        if (skipAttributes(position)) {
```

Comment:
Looks like this actually changes the filtering logic. We only want to apply `skipLimit` and `skipAttributes` for certain types of filters.

Reply:
Okay. Could you please elaborate? Why should those 2 _skips_ work only for some filters?

Reply:
There was a forum discussion about it if you want more context. Basically there are two categories of filtering. The ones that filter definitely invalid data (e.g. zero coordinates) and the ones that filter excessive data (e.g. duplicates). We always want to filter former, but want to keep latter if it has some important information (e.g. alarms).

Reply:
Done. I reverted to the original logic and also preserved almost the same order of filter checks (as it is in current master branch).